### PR TITLE
[DDW-853] Update electron

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR updated electron and related package versions.
[YouTrack Ticket](https://iohk.myjetbrains.com/youtrack/issue/DDW-853)

## Todos
- [x]  Update electron and its related packages to latest stable versions.
- [x]  Check `yarn.lock` to see if bad entires were added
- [x]  Check terminal to see if everything was installed properly
- [x]  Make sure that all `yarn ...` scripts work well
- [x] Fix "Blank screen fix" issues on macOS Mojave/Catalina
- [x] Fix "Wallet import" state folder selection
- [ ] ~Persist "Blank screen fix" active status after Daedalus restart~ - To be resolved in [DDW-155](https://jira.iohk.io/browse/DDW-155)

## Test Checklist
- [x]  Make sure all file open/save dialogs work well.
       * Go to `Wallet File Import` screen and check if file import works well.
       * Go to `Staking Rewards` screen and check if csv export works well.
       * Go to `Wallet Receive` screen and check if pdf download works well.
       * Go to `Paper Wallet Certificate` => `Instructions` dialog and check if print works well.
       * Open `Export Wallet to File` dialog and check if export works well.
       * Check if `Download Logs` works well.
- [x]  Make sure logs are printed out well on the console.
- [x]  Make sure all the user settings are saved well on settings.

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by tomothespian** [comment]
@yakovkaravelov I  have the same error as @DmitriiGaico. 
Can you check electron breaking changes? Maybe that can help you:
https://www.electronjs.org/docs/breaking-changes

**Comment by DmitriiGaico** [comment]
Linux Build not even started for me.
Thank you!

**Comment by nikolaglumac** [comment]
@cleverca22 @disassembler could you assist us with CI build failures in this PR? 🙏

**Comment by yakovkaravelov** [comment]
@nikolaglumac @daniloprates @DmitriiGaico @DominikGuzei Please review.
Thanks.

**Comment by gnpf** [comment]
@yakovkaravelov On Windows, when the system theme is "Dark", the menu bar in Daedalus is affected

image

This is not reproducible on productive builds (same theme on Win):

image

**Comment by nikolaglumac** [comment]
> @yakovkaravelov On Windows, when the system theme is "Dark", the menu bar in Daedalus is affected
> 
> image
> 
> This is not reproducible on productive builds (same theme on Win):
> 
> image

What is the problem here @gnpf?

**Comment by DmitriiGaico** [comment]
@nikolaglumac When in Windows Itself a Dark Theme is selected Daedalus also reads this and make the toolbar Also Dark. In Production FC build it has no affect. No matter what Windows theme is selected Daedalus toolbar is Pure White.

**Comment by nikolaglumac** [comment]
This is expected and it is not a bug. That's how new Electron works.

**Comment by nikolaglumac** [comment]
@yakovkaravelov @DominikGuzei I am seeing a lot of these log lines:

```
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
```

Any clue?

**Comment by IuliaDolishniak** [comment]
I am seeing the following issues so far:
1. After app reload all the wallets are removed
2. When I am trying to download logs there is no logs name predefined, then I press Cancel and after that when I am trying open it again dialog windows doesnt open

**Comment by nikolaglumac** [comment]
@DmitriiGaico blank screen fix is fixed by: https://github.com/input-output-hk/daedalus/pull/1887/commits/52bc6d823e7a366df37181f281d5b8e17cb8c93f

**Comment by yakovkaravelov** [comment]
> @yakovkaravelov @DominikGuzei I am seeing a lot of these log lines:
> 
> ```
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> (node:21285) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ELECTRON_LOG_CHANNEL-response listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
> ```
> 
> Any clue?

@nikolaglumac It's related to electron-log ipc communication. Will look into it.

**Comment by nikolaglumac** [comment]
@yakovkaravelov @daniloprates @tomothespian @DominikGuzei @IuliaDolishniak @ManusMcCole @gnpf @DmitriiGaico this fixes all the Electron dialog issues: https://github.com/input-output-hk/daedalus/pull/1887/commits/a6be1da8d40ba6c0dfe568444ec48f6886f8df4c
All of them are working now!

File: source/renderer/app/ipc/show-file-dialog-channels.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// @flow
```
[the remark this answers is not in the document]
Comment (nikolaglumac) [reply source/renderer/app/ipc/show-file-dialog-channels.js:1]: @daniloprates let's leave that for later 🙏 
Please create a card "Unify IPC communication patterns"

**Comment by nikolaglumac** [comment]
This commit fixed the blank screen fix issues: https://github.com/input-output-hk/daedalus/pull/1887/commits/fce2faef2e9c7ac4981d121d9362e38f37f13527

Blank screen fix ON:
Screenshot 2020-04-09 at 19 56 37

Blank screen fix OFF (default):
Screenshot 2020-04-09 at 19 57 14

**Comment by IuliaDolishniak** [comment]
> I am seeing the following issue so far:
> 
> 1. When I am trying to download logs there is no logs name predefined, then I press Cancel and after that when I am trying open it again dialog windows doesnt open
>    

Fixed

**Comment by yakovkaravelov** [comment]
> > nikolaglumac It's related to electron-log ipc communication. Will look into it.
> 
> @yakovkaravelov it's something related to the TransactionsStore, it seems:
> 
> image

@daniloprates It was `too many event emits triggered by logging`, fixed it.
Everything should be fine now, @nikolaglumac

**Comment by nikolaglumac** [comment]
@yakovkaravelov I am a bit surprised that you have reverted the logger IPC channel here: https://github.com/input-output-hk/daedalus/pull/1887/commits/5d09723bba95b1b41a8994474a0dd6d0b8db261d
Wasn't that the whole point of this upgrade? cc @DominikGuzei

**Comment by yakovkaravelov** [comment]
> @yakovkaravelov I am a bit surprised that you have reverted the logger IPC channel here: [5d09723](https://github.com/input-output-hk/daedalus/commit/5d09723bba95b1b41a8994474a0dd6d0b8db261d)
> Wasn't that the whole point of this upgrade? cc @DominikGuzei

We moved from global `electron-store` to ipc based `electron-store`.
Same for `Dialog`.
Because these packages were importing electron apis which were not accessible in new electron version.
Initially I did same for `daedalus-electron-log` as well, but I found following things lately
```
1. `daedalus-electron-log` works well as global utility even with new electron version which is not the case for above 2 packages.
2. If we use ipc for logging, `EventEmitter` memory leak happens because ipc channels have certain limit for number of events emitted.
We logs messages in tons of places in our app which is not ideal for this case.
Expanding limit can work, but I didn't want to change it for saving memory usage.
```
 
So eventually, I chose to restore the global logging package usage instead of ipc

**Comment by nikolaglumac** [comment]
@tomothespian I still see this test breaking :(

**Comment by tomothespian** [comment]
> @tomothespian I still see this test breaking :(

Fixed!

**Comment by nikolaglumac** [comment]
@DmitriiGaico does this happen on other builds too?

**Comment by DmitriiGaico** [comment]
@nikolaglumac This doesn't happens for Production FC2/SelfNode.
Trying previous build for this PR.

**Comment by DmitriiGaico** [comment]
@nikolaglumac Same Scroll Space in Build 12197

**Comment by nikolaglumac** [comment]
@cleverca22 it seems that Linux builds are not working at all. @DmitriiGaico just installed build  https://buildkite.com/input-output-hk/daedalus/builds/12220 and it doesn't even start. No logs... No nothing.

**Comment by DmitriiGaico** [comment]
@yakovkaravelov hey, unfortunately still reproducible. Latest Windows Build
image

**Comment by yakovkaravelov** [comment]
> @yakovkaravelov hey, unfortunately still reproducible. Latest Windows Build
> image

Fixed.

**Comment by nikolaglumac** [comment]
@DmitriiGaico @yakovkaravelov @DominikGuzei signing new Electron dylibs in the build process did the trick and we now have working "Blank screen fix" on all macOS versions 🎉 - thanks @johnalotoski for sorting this out!

This PR unfortunately still isn't ready to be merged as I have found one more issue:
Screenshot 2020-04-16 at 08 08 49

The launcher should persist the "--safe-mode" flag once we set it in Daedalus, but it looks like that is not the case anymore. This is most likely an issue in the launcher itself so we'll need help from @cleverca22 @disassembler or @johnalotoski to check it.

@DmitriiGaico I need you to check if this was also broke in the FC2 version:
1. Start Daedalus FC2
2. Activate "Blank screen fix" upon which Daedalus will restart into "Safe mode"
3. Exit Daedalus and start it again.
4. Check if the "Blank screen fix" is still active (it should still be active!)

**Comment by yakovkaravelov** [comment]
@DmitriiGaico Please review again. Thanks.

**Comment by DmitriiGaico** [comment]
@nikolaglumac Yes, it is broken in Prod FC2. After App Restart Blank Screen no longer active.

**Comment by DmitriiGaico** [comment]
@yakovkaravelov White Flashing Bug still reproducible in the Latest Build.

**Comment by yakovkaravelov** [comment]
@DmitriiGaico https://github.com/electron/electron/issues/18184
It is known issue of Electron 3+
It is said that latest VS Code Exploration builds fixed that issue which means new Electron release will fix it. (VS Code was made by using Electron)
So we can wait and see what it does.

**Pull request merged**